src/Domain/Encryption/Crypter.ts

```diff
@@ -0,0 +1,77 @@
+import { SNPureCrypto } from '@standardnotes/sncrypto-common'
+import { inject, injectable } from 'inversify'
+import TYPES from '../../Bootstrap/Types'
+import { User } from '../User/User'
+import { CrypterInterface } from './CrypterInterface'
+
+@injectable()
+export class Crypter implements CrypterInterface {
+  constructor (
+    @inject(TYPES.SNCrypto) private snCrypto: SNPureCrypto,
+    @inject(TYPES.ENCRYPTION_SERVER_KEY) private encryptionServerKey: string
+  ) {
+  }
+
+  async encryptForUser(value: string, user: User): Promise<string> {
+    const decryptedUserServerKey = await this.decryptUserServerKey(user)
+    const nonce = await this.snCrypto.generateRandomKey(192)
+
+    const encryptedValue = <string> await this.snCrypto.xchacha20Encrypt(
+      value,
+      nonce,
+      <string> decryptedUserServerKey,
+      ''
+    )
+
+    return this.formatEncryptedValue(User.ENCRYPTION_VERSION_1, encryptedValue, nonce)
+  }
+
+
+  async decryptForUser(value: string, user: User): Promise<string | null> {
+    const decryptedUserServerKey = await this.decryptUserServerKey(user)
+
+    const [ version, key, nonce ] = value.split(':')
```

Comment:
Should not be called `key`, but `ciphertext`. I think `key` may have been remnant from when decryption wasn't generic.